odl/contrib/theano/layer.py

```diff
@@ -83,12 +283,48 @@ def grad(self, inputs, output_grads):
             ran_weight = 1.0
 
         scale = dom_weight / ran_weight
-        if scale == 1.0:
-            adjoint_operator = self.operator.adjoint
-        else:
-            adjoint_operator = scale * self.operator.adjoint
 
-        return [TheanoOperator(adjoint_operator)(output_grads[0])]
+        op = self
+
+        class TheanoJacobian(theano.Op):
+
+            """Wrap ``op.derivative`` into a Theano Op.
+
+            This Op has two inputs, ``x`` and ``v``, where ``x``
+            is the point at which the Jacobian is taken, and ``v`` the
+            tensor to which it is applied. There is only one output,
+            which is of the same type as ``v`` (and ``x``).
+            """
+
+            def make_node(self, x, v):
+                """Create a node for the computation graph."""
+                x = theano.tensor.as_tensor_variable(x)
+                v = theano.tensor.as_tensor_variable(v)
+                return theano.Apply(self, [x, v], [x.type()])
+
+            def perform(self, node, inputs_storage, output_storage):
+                """Evaluate this node's computation.
+
+                This method computes ::
+
```

Comment:
Remove blankline

Reply:
That would ruin the typesetting, keeping it as-is.